### ISSUE TITLE
NVDA doesn't read the whole content of the "Confirm deletion" dialog in Win7

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#2112]
**Reported by k_kolev1985 on 2012-02-18 18:53**
NVDA does not read the whole content of the "Confirm deletion" dialog for shortcuts for applications on the desktop (maybe even in other places as well) under Windows 7.

Steps to reproduce:
1. Load NVDA (I'm using main snapshot 5012, but it happens with 2011.3 as well).
2. Go to the desktop.
3. Find a shortcut for an installed application (even the one for an installed copy of NVDA should do it).
4. Press the "Delete" key to invoke the "Delete" command for the selected element (the shortcut in our case).
5. The "Confirm deletion" dialog appears. NVDA should read its whole content, but it skips the upper part where Windows informs us that the deletion of the shortcut will not remove the correspondent application from our system, but only the shortcut itself.

This is tested with NVDA 2011.3 and main snapshot #5012 under Windows 7 Ultimate 32-bit with bulgarian localization and regional and language settings set to bulgarian.

**Comment by bhavyashah** [comment]
@k-kolev1985 Could you please test and report whether or not the reported issue still persists? Even if it does, I expect priority to be P4, right? @feerrenrut

**Comment by k-kolev1985** [comment]
Yes, this still occurs under Windows 7 with NVDA 2017.2.

**Comment by feerrenrut** [comment]
I cant get this dialog to show with Windows 7 Home Premium. 

@k-kolev1985 Could you please add both the full expected text to this issue, and the actual text reported by NVDA to this issue? So that I can understand better how you are approaching this issue, could you also tell me how you read the text which is not initially reported by NVDA. Thanks!

**Comment by k-kolev1985** [comment]
@feerrenrut What - you don't get the deletion confirmation dialog for a shortcut? Have you by any chance disabled the deletion confirmation dialog in the properties of the "Recycle Bin"?

Anyway, here is what I get NVDA to read when the dialog appears and manually with screen review, both in bulgarian and my approximate translation to english:

// This is what NVDA reads when the dialog appears, in bulgarian:
Изтриване на пряк път  диалог  Програми и компоненти
NVDA
Местоположение: nvda_slave (C:\Програмни файлове\NVDA)
Да  бутон  Alt+д

// This is what NVDA reads when the dialog appears, in english (my approximate translation):
Delete a shortcut dialog Programs and Features
NVDA
Location: nvda_slave (C:\Program Files\NVDA)
Yes button Alt+Y

// This is what NVDA reads from the dialog with screen review from top to bottom, in bulgarian:
Изтриване на пряк път
Наистина ли искате да преместите този пряк път в "Кошчето"?

Изтриването на прекия път не деинсталира програмата. Toва само премахва иконата, която сочи към програмата. Ако искате да деинсталирате програмата, използвайте Програми и компоненти.
// I don't visually see the next sentence on the screen, but NVDA sees it and speaks it.
Трябва да потвърдите тази операция.
NVDA
Местоположение: nvda_slave (C:\Програмни файлове\NVDA)
празен
Да Не

// This is what NVDA reads when the dialog appears, in english (my approximate translation):
Delete a shortcut
Do you realy want to move this shortcut to the "Recycle bin"?

The deletion of the shortcut does not uninstall the program. This only removes the icon that points to the program. If you want to uninstall the program, use Programs and Features.
// I don't visually see the next sentence on the screen, but NVDA sees it and speaks it.
You must confirm this operation.
NVDA
Location: nvda_slave (C:\Program Files\NVDA)
blank
Yes No

I can provide screenshots as well, if needs be.

**Comment by feerrenrut** [comment]
I have, at some point, seen the dialog you are talking about, but can't reproduce it on my current Windows 7 Home Premium install (fresh Virtual Machine). I only get the text "Are you sure you want to move this shortcut to the recycle bin?" followed by the path to the shortcut. 
To double check we are doing the same thing:
1. Go to desktop
2. Highlight shortcut icon (I am choosing Firefox)
3. Press the delete key on the keyboard.

Perhaps this feature is not available in Home Premium, I cant find any information on how to enable it.

**Comment by k-kolev1985** [comment]
Yes, we're doing the same thing, although I'm using the NVDA shortcut for the purpose. There may be at least 2 reasons for you not being able to reproduce it:
1. The additional prompt for "Programs and Features" is not available on Windows 7 Home Premium.
2. Are you signed as an administrator? I am. The prompt may be available only for administrative accounts.

**Comment by k-kolev1985** [comment]
Yes, the bug is still reproducible in Windows 7 with latest stable NVDA 2018.3.2.

**Comment by Adriani90** [comment]
@k-kolev1985 I don'T think this issue disappeared in the meantime, but could you please test now with NVDA last alpha version? Just to keep the developers updated if this has been fixed or not meanwhile.

**Comment by k-kolev1985** [comment]
@Adriani90 Yes, the bug is still reproducible with the latest alpha build (alpha-20097,6a573cd9) of NVDA.

**Comment by Adriani90** [comment]
I am closing this issue as stale, since Windows 7 support has completely ended in 2020 and I don't expect any futher updates to it.
